kbase-extension/static/kbase/js/common/jobStateViewer.js

```diff
@@ -0,0 +1,324 @@
+/**
+ *
+ * Usage:
+ * const viewer = new JobStateViewer({
+ *      jobManager,
+ *      showHistory: true,
+ * });
+ * viewer.start({
+ *     jobId: <some job id>,
+ *     node: <a DOM node>
+ * })
+ *
+ * viewer.stop()
+ */
+define([
+    'bluebird',
+    'common/runtime',
+    'common/props',
+    'common/ui',
+    'common/fsm',
+    'common/jobs',
+    'common/jobCommMessages',
+    'common/html',
+    'common/runClock',
+    'common/errorDisplay',
+    'util/developerMode',
+], (Promise, Runtime, Props, UI, Fsm, Jobs, jcm, html, RunClock, ErrorDisplay, devMode) => {
+    'use strict';
+
+    const t = html.tag,
+        div = t('div'),
+        span = t('span'),
+        cssBaseClass = 'kb-log',
+        stateCssBaseClass = 'kb-job-state-viewer';
+
+    /**
+     * The entrypoint to this widget. This creates the job state viewer and initializes it.
+     * Starting it is left as a lifecycle method for the caller.
+     *
+     * config options:
+     *  showHistory   {boolean} - whether job status lines should include history
+     *                            (queue and run times)
+     *  developerMode {boolean} - whether or not to enable developer mode, which
+     *                            prints out extra debugging information
+     */
+
+    class JobStateViewerCore {
```

Comment:
created by picking out all the job status line-related bits from the JobLogViewer module